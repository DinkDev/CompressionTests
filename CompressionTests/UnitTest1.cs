using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace CompressionTests
{
    using System;
    using System.Threading;
    using System.Threading.Tasks;
    using ApprovalTests;
    using ApprovalTests.Reporters;
    using Microsoft.Extensions.Logging;

    [TestClass]
    [UseReporter(typeof(WinMergeReporter))]
    public class UnitTest1
    {
        public TestContext TestContext { get; set; }

        [TestMethod]
        public async Task OriginalCompleteServerTest1()
        {
            var tokenSource = new CancellationTokenSource();
            var loggerFake = new LoggerFake(nameof(OriginalCompleteServerTest1));
            Stubs.Stream.Logger = (ILogger)loggerFake;
            var sut = new CompleteServerStub(tokenSource.Token, () => tokenSource.Cancel());
            sut.Logger = loggerFake;
            sut.CurrentDateTime = () => new DateTime(2021, 6, 28, 01, 0, 0);

            await sut.ExecuteAsync();

            loggerFake.LogTrace("Test completed!");

            var output = loggerFake.LoggedData.ToString();

            Approvals.Verify(output);
        }

        [TestMethod]
        public async Task RefactoredCompleteServerTest1()
        {
            var tokenSource = new CancellationTokenSource();
            var loggerFake = new LoggerFake(nameof(OriginalCompleteServerTest1));
            Stubs.Stream.Logger = (ILogger)loggerFake;
            var sut = new RefactoredCompleteServerStub(tokenSource.Token, () => tokenSource.Cancel());
            sut.Logger = loggerFake;
            sut.CurrentDateTime = () => new DateTime(2021, 6, 28, 02, 0, 0);

            await sut.ExecuteAsync();

            loggerFake.LogTrace("Test completed!");

            var output = loggerFake.LoggedData.ToString();

            Approvals.Verify(output);
        }
    }
}
