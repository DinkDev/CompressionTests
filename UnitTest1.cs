using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace CompressionTests
{
    using System.Threading;
    using System.Threading.Tasks;
    using Stubs;

    [TestClass]
    public class UnitTest1
    {
        public TestContext TestContext { get; set; }

        [TestMethod]
        public async Task TestMethod1()
        {
            var tokenSource = new CancellationTokenSource();
            Stubs.Stream.TestContext = TestContext;
            var sut = new CompleteServerStub(tokenSource.Token, () => tokenSource.Cancel());
            sut.Logger = new fm.Extensions.Logging.TestContext.TestContextLogger(nameof(TestMethod1), TestContext);
            
            await sut.ExecuteAsync();

            TestContext.WriteLine($"{nameof(UnitTest1)}.{nameof(TestMethod1)} completed!");
        }
    }
}
