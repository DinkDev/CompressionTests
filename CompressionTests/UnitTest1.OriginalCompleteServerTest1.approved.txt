OriginalCompleteServerTest1: Trace: CompleteServer.ExecuteAsync - Starting at 6/28/2021 1:00:00 AM
OriginalCompleteServerTest1: Information: Processing starting - bundle: 1CLM1IDS, package: 13060495, application number: 13060495, Documents: HP49ZKJ2PXXIFW3
OriginalCompleteServerTest1: Debug: CompleteServer.ProcessOutQueuePackagesAsync: Starting bundle ID: 33, Timeout: 6/28/2021 1:05:00 AM
OriginalCompleteServerTest1: Information: Validated package 13060495
OriginalCompleteServerTest1: Information: MemoryStream: Close - start data dump.
OriginalCompleteServerTest1: Information: TarStream:
OriginalCompleteServerTest1: Information: ZipStream:
OriginalCompleteServerTest1: Information: SetLevel: 3
OriginalCompleteServerTest1: Information: PutNextEntry: ZipEntry: 13060495.2013-12-12.HP49ZKJ2PXXIFW3\13060495.2013-12-12.HP49ZKJ2PXXIFW3.XML, 6/28/2021 1:00:00 AM, 65
OriginalCompleteServerTest1: Information: Write: Contents from 13060495.2013-12-12.HP49ZKJ2PXXIFW3_Transformed.xml, 0, 65
OriginalCompleteServerTest1: Information: CloseEntry
OriginalCompleteServerTest1: Information: IsStreamOwner: False
OriginalCompleteServerTest1: Information: Close
OriginalCompleteServerTest1: Information: Close
OriginalCompleteServerTest1: Information: MemoryStream: Close - end data dump.
OriginalCompleteServerTest1: Information: CompleteServer.ProcessOutQueuePackagesAsync : Package 13060495 sent.
OriginalCompleteServerTest1: Trace: CompleteServer.SendBundleAsync: 
OriginalCompleteServerTest1: Information: FileStream: Close - start data dump.
OriginalCompleteServerTest1: Information: TarStream:
OriginalCompleteServerTest1: Information: Close
OriginalCompleteServerTest1: Information: Close
OriginalCompleteServerTest1: Information: ZipStream:
OriginalCompleteServerTest1: Information: FileStream: Close - end data dump.
OriginalCompleteServerTest1: Information: FileStream: Close - start data dump.
OriginalCompleteServerTest1: Information: TarStream:
OriginalCompleteServerTest1: Information: Close
OriginalCompleteServerTest1: Information: Close
OriginalCompleteServerTest1: Information: ZipStream:
OriginalCompleteServerTest1: Information: FileStream: Close - end data dump.
OriginalCompleteServerTest1: Trace: Created Tar: TemporaryDirectory\33.TAR at 6/28/2021 1:00:00 AM
OriginalCompleteServerTest1: Information: CompleteServer.SendBundleOutAsync: bundle file name: TemporaryDirectory\33.TAR
OriginalCompleteServerTest1: Information: CompleteServer.SendBundleOutAsync: copied bundle to save: TemporaryDirectory\33.TAR.SENT
OriginalCompleteServerTest1: Information: CompleteServer.SendBundleAsync: new bundle started 10101
OriginalCompleteServerTest1: Trace: Test completed!
